### ISSUE TITLE
Memory Leak (❎ Apple CoreML, ☑️ Nvidia Cuda)

### ISSUE THREAD

**Issue by hugo4711** [s0md3v/roop#321]
I run roop on a Mac M1. I have installed onnxruntime-silicon.

Everything is working fine, until the system crashes because of the system running out of memory.

I have a target video that is 1280x720 in size, around 1 minute in length.

There are 6 python-3.10 processes runing in parallel.

Memory for each process increases from 2 Gbyte per Process to 7.5 Gbyte per Process at around 25% of the work.

**Comment by Rigo-m** [comment]
Same here. CPU processing runs correctly in the `next` branch.

**Comment by abuvanth** [comment]
![Screenshot_2023-06-04_20-00-34](https://github.com/s0md3v/roop/assets/20999104/bf01b2d4-0aa5-4d4a-a12f-fa42c8f51fe5)
same here

**Comment by hugo4711** [comment]
I got it to work with --max-cpu 1

But then the process just stops at around 25%

**Comment by hugo4711** [comment]
Currently trying the next branch. Looks good!

**Comment by Rigo-m** [comment]
> Currently trying the next branch. Looks good!

Looks good also with `--gpu-vendor apple`?

**Comment by Sgoldik** [comment]
Branch - next.:
`python run.py --gpu-vendor=apple --max-memory=20000 --gpu-threads=1`

<img width="390" alt="Screenshot 2023-06-04 at 23 05 53" src="https://github.com/s0md3v/roop/assets/10558892/951ac2dc-d856-450e-8e33-0a7022b63a98">

**Comment by hugo4711** [comment]
> Branch - next.: `python run.py --gpu-vendor=apple --max-memory=20000 --gpu-threads=1`
> 
> <img alt="Screenshot 2023-06-04 at 23 05 53" width="390" src="https://user-images.githubusercontent.com/10558892/243195272-951ac2dc-d856-450e-8e33-0a7022b63a98.png">

That doesn't work for me:

```
Applied providers: ['CoreMLExecutionProvider', 'CPUExecutionProvider'], with options: {'CPUExecutionProvider': {}, 'CoreMLExecutionProvider': {}}
inswapper-shape: [1, 3, 128, 128]
Exception in thread Thread-1 (<lambda>):
Traceback (most recent call last):
  File "/Users/nolte/anaconda3/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/Users/nolte/anaconda3/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/nolte/Documents/roop/roop/ui.py", line 109, in <lambda>
    set_preview_handler(lambda: preview_thread(lambda: test_preview(create_test_preview)))
  File "/Users/nolte/Documents/roop/roop/ui.py", line 103, in test_preview
    frame = create_test_preview(preview_frame_slider.get())
  File "/Users/nolte/Documents/roop/roop/core.py", line 237, in create_test_preview
    return process_faces(
  File "/Users/nolte/Documents/roop/roop/swapper.py", line 34, in process_faces
    target_frame = swap_face_in_frame(source_face, face, target_frame)
  File "/Users/nolte/Documents/roop/roop/swapper.py", line 25, in swap_face_in_frame
    return get_face_swapper().get(frame, target_face, source_face, paste_back=True)
  File "/Users/nolte/anaconda3/lib/python3.10/site-packages/insightface/model_zoo/inswapper.py", line 50, in get
    latent = source_face.normed_embedding.reshape((1,-1))
AttributeError: 'NoneType' object has no attribute 'normed_embedding'
Exception in thread Thread-2 (<lambda>):
Traceback (most recent call last):
  File "/Users/nolte/anaconda3/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/Users/nolte/anaconda3/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/nolte/Documents/roop/roop/ui.py", line 109, in <lambda>
    set_preview_handler(lambda: preview_thread(lambda: test_preview(create_test_preview)))
  File "/Users/nolte/Documents/roop/roop/ui.py", line 103, in test_preview
    frame = create_test_preview(preview_frame_slider.get())
  File "/Users/nolte/Documents/roop/roop/core.py", line 237, in create_test_preview
    return process_faces(
  File "/Users/nolte/Documents/roop/roop/swapper.py", line 34, in process_faces
    target_frame = swap_face_in_frame(source_face, face, target_frame)
  File "/Users/nolte/Documents/roop/roop/swapper.py", line 25, in swap_face_in_frame
    return get_face_swapper().get(frame, target_face, source_face, paste_back=True)
  File "/Users/nolte/anaconda3/lib/python3.10/site-packages/insightface/model_zoo/inswapper.py", line 50, in get
    latent = source_face.normed_embedding.reshape((1,-1))
AttributeError: 'NoneType' object has no attribute 'normed_embedding'
```

**Comment by hugo4711** [comment]
Ok, double checked it: That is another issue related to the "Face" Image I chose...

The GPU-Command line seems to work also!

**Comment by henryruhs** [comment]
Issues for the next branch are kind of sub-optimal as this branch is under active development and changes every hour. Still working on the memory leak.

**Comment by henryruhs** [comment]
Memory leak by tensorflow fixed... I now get 60fsp with an 4090 and 16 GPU threads.

**Comment by abuvanth** [comment]
pretty fast 50 fps with 3080ti and 16 GPU threads

**Comment by henryruhs** [comment]
My tensorflow fix does not translate to Apple Silicon. I reopen it.

**Comment by Mayorc1978** [comment]
> Memory leak by tensorflow fixed... VRAM is now free for Cuda! I get 60fsp with an 4090 and 16 GPU threads.

The TensorFlow requirement wasn't there in the first Roop versions, what is being used for in the latest branches?

**Comment by Rigo-m** [comment]
@henryruhs did you change the title because apple silicon got fixed?

**Comment by henryruhs** [comment]
Nope, the CPU multi-processing seems to be platform independent.

**Comment by greentornado** [comment]
<img width="559" alt="Screen Shot 2023-06-08 at 10 06 22" src="https://github.com/s0md3v/roop/assets/280004/bbe6b29f-95f9-4042-a8b8-7b108c31246e">

I got this error on main branch 
run command: python run.py --gpu-vendor=apple  --gpu-threads=1 --max-memory=1000

**Comment by henryruhs** [comment]
Can you try if next branch works better for you?

**Comment by chenull** [comment]
> Can you try if next branch works better for you?

using `next` branch, still got the error

```
Applied providers: ['CoreMLExecutionProvider', 'CPUExecutionProvider'], with options: {'CPUExecutionProvider': {}, 'CoreMLExecutionProvider': {}}
find model: /Users/ayik/.insightface/models/buffalo_l/w600k_r50.onnx recognition ['None', 3, 112, 112] 127.5 127.5
set det-size: (640, 640)
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1072/1072 [00:01<00:00, 801.65it/s]
Status: Creating temp resources...
Status: Extracting frames...
Status: Swapping in progress...
Processing:   0%|                                                                                                                   | 0/1072 [00:00<?, ?frame/s]Applied providers: ['CoreMLExecutionProvider', 'CPUExecutionProvider'], with options: {'CPUExecutionProvider': {}, 'CoreMLExecutionProvider': {}}
Processing:   0%|▏                                                                                                        | 2/1072 [00:12<1:50:29,  6.20s/frame]inswapper-shape: [1, 3, 128, 128]
Processing:  23%|████████████████████████▌                                                                                | 251/1072 [02:22<10:10,  1.35frame/s][1]    61111 killed     python run.py --gpu-vendor=apple -t  -f  -o
/opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
```

killed around frame 250 out of 1075

the CLI argument was

```
python run.py --gpu-vendor=apple -t ttt -f fff -o ooo
```

**Comment by henryruhs** [comment]
I fixed the different memory unit for mac but still had trouble with that. Probably something is off with onxxruntime-silicon?

**Comment by aqgasa** [comment]
Yes, this solution seems to work. I’ll do some more tests and be back with the details soon.

**Comment by henryruhs** [comment]
@aqgasa You managed to compile it?

**Comment by aqgasa** [comment]
No, I didn't. I tested with this branch https://github.com/microsoft/onnxruntime/tree/edgchen1/fix_mac_build_issues_fix

There's no memory leak, but unfortunately the faces are all messed up after the face swap.

**Comment by henryruhs** [comment]
What does messed up mean?

**Comment by oneafteromega** [comment]
> What does messed up mean?

I did something similar, using https://github.com/microsoft/onnxruntime/tree/edgchen1/fix_mac_build_issues_fix to compile a new onnxruntime-silicon, but the face swapping effect is not working correctly.
Following are two preview images, one converted using coreml and the other using CPU.

![Snipaste_2023-06-10_14-08-56](https://github.com/s0md3v/roop/assets/49817323/a6cf21da-25c4-4a80-8311-47204826345b)
![Snipaste_2023-06-10_14-09-47](https://github.com/s0md3v/roop/assets/49817323/9d1989b3-af94-41f2-8855-b8ffe95f3cd4)

**Comment by aqgasa** [comment]
> What does messed up mean?

Sorry I didn't explain it properly. The images were like in @oneafteromega 's example above.

**Comment by henryruhs** [comment]
Yes, the hair looks kind of weird... 😁

**Comment by aqgasa** [comment]
I merged https://github.com/microsoft/onnxruntime/tree/edgchen1/fix_mac_build_issues_fix and https://github.com/cansik/onnxruntime-silicon, and was able to build it (it wasn't pretty).

I'd say its more like a proof of concept kind of thing. I haven't had much time to test it, but it seems to work.

I can upload the wheel somewhere if someone wants to try it.

**Comment by henryruhs** [comment]
can you release a fixed version on pypi?

**Comment by aqgasa** [comment]
I've never done that. I have to look into it.

**Comment by henryruhs** [comment]
@aqgasa There is always a first time, I guess you never compiled a runtime before either. Good luck!

**Comment by Rigo-m** [comment]
> I merged https://github.com/microsoft/onnxruntime/tree/edgchen1/fix_mac_build_issues_fix and https://github.com/cansik/onnxruntime-silicon, and was able to build it (it wasn't pretty).
> 
> I'd say its more like a proof of concept kind of thing. I haven't had much time to test it, but it seems to work.
> 
> I can upload the wheel somewhere if someone wants to try it.

Do you have any instructions on how to build it correctly? I've tried by forking onnxruntime-silicon but github action fails

**Comment by sphuff** [comment]
I manually built the same branch as @aqgasa, but still seeing the blurry face issue above. 

@aqgasa , did you use any custom build arguments different than those specified in [onnxruntime-silicon](https://github.com/cansik/onnxruntime-silicon/blob/main/build-macos.sh)?

**Comment by cansik** [comment]
I am not involved in the roop project, but just wanted to let you know that I have released a new version of [onnxruntime-silicon](https://github.com/cansik/onnxruntime-silicon) ([v1.14.1](https://github.com/cansik/onnxruntime-silicon/releases/tag/v1.14.1) and [v1.15.0](https://github.com/cansik/onnxruntime-silicon/releases/tag/v1.15.0)). Maybe this will help with the CoreML memory leak problem.

**Comment by h0x91b** [comment]
Thanks @cansik!
The memory leak is fixed, but the face is blurry...
![Screenshot 2023-06-12 at 21 16 11](https://github.com/s0md3v/roop/assets/810011/1e63cca2-ef8b-4a4d-baa1-1dde3430b4fb)

**Comment by h0x91b** [comment]
1.14.1 - faces swap works well, but memory leak is still there

**Comment by verback2308** [comment]
> I am not involved in the roop project, but just wanted to let you know that I have released a new version of [onnxruntime-silicon](https://github.com/cansik/onnxruntime-silicon) ([v1.14.1](https://github.com/cansik/onnxruntime-silicon/releases/tag/v1.14.1) and [v1.15.0](https://github.com/cansik/onnxruntime-silicon/releases/tag/v1.15.0)). Maybe this will help with the CoreML memory leak problem.

Hi, could you build a 1.14.1 version from [my fork](https://github.com/verback2308/onnxruntime/tree/bugfix/coreml-memoryleak-1.14.1)? (it is the same as 1.14.1, but with a memory leak fix).

Thank you.

**Comment by jet3004** [comment]
Can someone _definitively_ say what arguments M1/M2 users should be using now to launch roop and produce the fastest possibly speeds? It works well enough but I know I'm leaving something on the table and can't get the current answer between here and Discord.

Is it _python3 run.py --execution-provider=coreml_ ?

**Comment by henryruhs** [comment]
@jet3004 You should join us on Discord, we have a helpful community. You can try to use `--execution-threads` too.

**Comment by h0x91b** [comment]
@henryruhs can you share please a link to Discord server?

**Comment by henryruhs** [comment]
There is a pinned issue!

**Comment by jet3004** [comment]
> @jet3004 You should join us on Discord, we have a helpful community. You can try to use `--execution-threads` too.

thanks, I did look there, in fact...admittedly, I'm newer to Python so learning it one by one. So with the latest updates, would you do something like this for the best performance...? 

_python3 run.py --execution-provider=coreml --execution-threads=1_ 

Running tests now.

**Comment by Anson2048** [comment]
[onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl.zip](https://github.com/s0md3v/roop/files/11761380/onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl.zip)

Unzip the file and then install onnxruntime_silicon-1.14.2 using the command 

```bash
pip uninstall onnxruntime-silicon
pip install ./onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl
```

This version will solve the issues of memory leaks and blurriness after face swapping.

**Comment by guangmean** [comment]
> 
I get the same error on next branch when run roop on M1 with the following command:
python run.py --execution-provider coreml

zsh: killed     python run.py --execution-provider coreml       | 92/549 [00:54<04:37,  1.64frame/s, execution_providers=['CoreMLExecutionProvider'], threads=1, memory=2]
roop/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '

**Comment by jet3004** [comment]
> [onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl.zip](https://github.com/s0md3v/roop/files/11761380/onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl.zip)
> 
> Unzip the file and then install onnxruntime_silicon-1.14.2 using the command
> 
> ```shell
> pip uninstall onnxruntime-silicon
> pip install ./onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl
> ```
> 
> This version will solve the issues of memory leaks and blurriness after face swapping.

Thanks for this, however, I get this error (and see others are getting the same) each time:

```
Traceback (most recent call last):
  File "/Users/jet/roop/run.py", line 3, in <module>
	from roop import core
  File "/Users/jet/roop/roop/core.py", line 21, in <module>
	import roop.globals
  File "/Users/jet/roop/roop/globals.py", line 1, in <module>
	import onnxruntime
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/__init__.py", line 55, in <module>
	raise import_capi_exception
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/__init__.py", line 23, in <module>
	from onnxruntime.capi._pybind_state import (
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/_pybind_state.py", line 33, in <module>
	from .onnxruntime_pybind11_state import *  # noqa
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so, 0x0002): Library not loaded: @rpath/libre2.9.dylib
  Referenced from: <2915FFF9-219F-3427-85D2-50EA30F61B4B> /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so
  Reason: tried: '/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/libre2.9.dylib' (no such file), '/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/libre2.9.dylib' (no such file), '/usr/lib/libre2.9.dylib' (no such file, not in dyld cache)

```
I can uninstall _onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl_ and _pip install onnxruntime-silicon_  and it launches without the error above but, using installing your _onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl_ produces the above every time.

**Comment by pashanskiy** [comment]
> > [onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl.zip](https://github.com/s0md3v/roop/files/11761380/onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl.zip)
> > Unzip the file and then install onnxruntime_silicon-1.14.2 using the command
> > ```shell
> > pip uninstall onnxruntime-silicon
> > pip install ./onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This version will solve the issues of memory leaks and blurriness after face swapping.
> 
> Thanks for this, however, I get this error (and see others are getting the same) each time:
> 
> ```
> Traceback (most recent call last):
>   File "/Users/jet/roop/run.py", line 3, in <module>
> 	from roop import core
>   File "/Users/jet/roop/roop/core.py", line 21, in <module>
> 	import roop.globals
>   File "/Users/jet/roop/roop/globals.py", line 1, in <module>
> 	import onnxruntime
>   File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/__init__.py", line 55, in <module>
> 	raise import_capi_exception
>   File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/__init__.py", line 23, in <module>
> 	from onnxruntime.capi._pybind_state import (
>   File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/_pybind_state.py", line 33, in <module>
> 	from .onnxruntime_pybind11_state import *  # noqa
> ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so, 0x0002): Library not loaded: @rpath/libre2.9.dylib
>   Referenced from: <2915FFF9-219F-3427-85D2-50EA30F61B4B> /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so
>   Reason: tried: '/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/libre2.9.dylib' (no such file), '/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/libre2.9.dylib' (no such file), '/usr/lib/libre2.9.dylib' (no such file, not in dyld cache)
> ```
> 
> I can uninstall _onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl_ and _pip install onnxruntime-silicon_ and it launches without the error above but, using installing your _onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl_ produces the above every time.

I had the same problem and solved it in the most stupid way:

```shell
brew install re2
cp /opt/homebrew/lib/libre2.10.dylib /opt/homebrew/lib/libre2.9.dylib
``` 

After that everything works fine, no artifacts or memory leaks

**Comment by guangmean** [comment]
> > > [onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl.zip](https://github.com/s0md3v/roop/files/11761380/onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl.zip)
> > > Unzip the file and then install onnxruntime_silicon-1.14.2 using the command
> > > ```shell
> > > pip uninstall onnxruntime-silicon
> > > pip install ./onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > This version will solve the issues of memory leaks and blurriness after face swapping.
> > 
> > 
> > Thanks for this, however, I get this error (and see others are getting the same) each time:
> > ```
> > Traceback (most recent call last):
> >   File "/Users/jet/roop/run.py", line 3, in <module>
> > 	from roop import core
> >   File "/Users/jet/roop/roop/core.py", line 21, in <module>
> > 	import roop.globals
> >   File "/Users/jet/roop/roop/globals.py", line 1, in <module>
> > 	import onnxruntime
> >   File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/__init__.py", line 55, in <module>
> > 	raise import_capi_exception
> >   File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/__init__.py", line 23, in <module>
> > 	from onnxruntime.capi._pybind_state import (
> >   File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/_pybind_state.py", line 33, in <module>
> > 	from .onnxruntime_pybind11_state import *  # noqa
> > ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so, 0x0002): Library not loaded: @rpath/libre2.9.dylib
> >   Referenced from: <2915FFF9-219F-3427-85D2-50EA30F61B4B> /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so
> >   Reason: tried: '/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/libre2.9.dylib' (no such file), '/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/onnxruntime/capi/libre2.9.dylib' (no such file), '/usr/lib/libre2.9.dylib' (no such file, not in dyld cache)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I can uninstall _onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl_ and _pip install onnxruntime-silicon_ and it launches without the error above but, using installing your _onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl_ produces the above every time.
> 
> I had the same problem and solved it in the most stupid way:
> 
> ```shell
> brew install re2
> cp /opt/homebrew/lib/libre2.10.dylib /opt/homebrew/lib/libre2.9.dylib
> ```
> 
> After that everything works fine, no artifacts or memory leaks
 
**Thanks for these and it works very well** 👍👍👍👍👍👍

Run the following command after uninstall onnxruntime-silicon & reinstall onnxruntime_silicon-1.14.2-cp310-cp310-macosx_11_0_arm64.whl

`python run.py --execution-provider coreml`

I get the following libre2.9.dyld not found error:

```
Traceback (most recent call last):
  File "/Users/abc/roop/run.py", line 3, in <module>
    from roop import core
  File "/Users/abc/roop/roop/core.py", line 17, in <module>
    import onnxruntime
  File "/Users/abc/miniconda3/envs/roop/lib/python3.10/site-packages/onnxruntime/__init__.py", line 55, in <module>
    raise import_capi_exception
  File "/Users/abc/miniconda3/envs/roop/lib/python3.10/site-packages/onnxruntime/__init__.py", line 23, in <module>
    from onnxruntime.capi._pybind_state import (
  File "/Users/abc/miniconda3/envs/roop/lib/python3.10/site-packages/onnxruntime/capi/_pybind_state.py", line 33, in <module>
    from .onnxruntime_pybind11_state import *  # noqa
ImportError: dlopen(/Users/abc/miniconda3/envs/roop/lib/python3.10/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so, 0x0002): Library not loaded: @rpath/libre2.9.dylib
  Referenced from: <2915FFF9-219F-3427-85D2-50EA30F61B4B> /Users/abc/miniconda3/envs/roop/lib/python3.10/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so
  Reason: tried: '/Users/abc/miniconda3/envs/roop/lib/python3.10/site-packages/onnxruntime/capi/libre2.9.dylib' (no such file), '/Users/abc/miniconda3/envs/roop/lib/python3.10/site-packages/onnxruntime/capi/libre2.9.dylib' (no such file), '/Users/abc/miniconda3/envs/roop/bin/../lib/libre2.9.dylib' (no such file), '/Users/abc/miniconda3/envs/roop/bin/../lib/libre2.9.dylib' (no such file), '/usr/local/lib/libre2.9.dylib' (no such file), '/usr/lib/libre2.9.dylib' (no such file, not in dyld cache)
```

But I just make a libre2.9.dylib link to /Users/abc/miniconda3/envs/roop/lib/libre2.9.dylib and it worked.
Hope this can help others.

```
cd ~/miniconda3/envs/roop/lib

ln -s /opt/homebrew/lib/libre2.9.dylib libre2.9.dylib
```

**Comment by henryruhs** [comment]
Guys, please join discord and resolve your platform issues there - otherwise I will lock this thread.

**Comment by henryruhs** [comment]
As multi processing has been removed, the memory leak for Apple Silicon should be gone.

We agree that the onnxruntime-silicon needs an update but that cannot be resolved from our side. Therefore I close that issue. In case there are still Apple Silicon related issues, join us on Discord for help.
